### PR TITLE
fixed the gate range default

### PR DESCRIPTION
**Pull request by mts299**
# Scope 

Resolving the FOV being too small for grid plotting 

**issue**: #116

## Approval

**Number of approvals:** 1 approval easy fix

# Test

```
import matplotlib.pyplot as plt 
import pydarn

#Read in fitACF file
grid_file = "20170729.0202.00.fhw.grid"
grid_data = pydarn.SuperDARNRead(grid_file).read_grid()

pydarn.Grid.plot_grid(grid_data)
plt.show()
```

Figure_1

### CODE REVIEW

**Comment by Devin-Huyghebaert** [comment]
Tested with:

> 
> import matplotlib.pyplot as plt 
> import datetime as dt
> import pydarn
> 
> #Read in fitACF file
> grid_file = "data/20150318.0002.00.sas.grid"
> grid_data = pydarn.SuperDARNRead(grid_file).read_grid()
> 
> fanplot = pydarn.Grid.plot_grid(grid_data, start_time=dt.datetime(2015,3,18,0,30),parameter='vel')
> plt.show()

Gave error:

> grid.py, line 160, in plot_grid
>     _, aacgm_lons, _, _, ax = Fan.plot_fov(stid, dtime,
> fan.py, line 308, in plot_fov
>     plt.polar(thetas[ranges[1] - 1, 0:thetas.shape[1] - 1],
> IndexError: index 224 is out of bounds for axis 0 with size 76

Worked with fhw (20170729.0002.00.fhw.grid) and mcm (20161124.0001.00.mcm.a.grid) files I tried though.

**Comment by mts299** [comment]
Ha you beat me but yes I just found this bug as well. fixing it now.

**Comment by billetd** [comment]
I would like this to be put on hold for now, as it seems like a bandage over the actual issue. This seems to be that the radar FOV files themselves have a mis-match with the hardware files.

Also, in the example in the main comment, FHW is used which goes up to 110 range gates in both the HDW and radar FOV files. Was the issue simply that it defaulted to only showing 75 ranges in `plot_fov`?

File: pydarn/plotting/fan.py [hunk 1]
```diff
@@ -268,6 +270,9 @@ def plot_fov(cls, stid: str, dtime: dt.datetime, ax=None,
             rs - radius polar coordinates
         """
         # Get radar beam/gate locations
+        if ranges == []:
+            max_ranges = SuperDARNRadars.radars[stid].hardware_info.gates
+            ranges = [0, max_ranges]
```
Comment (billetd) [inline pydarn/plotting/fan.py:275]: This part (L273-275) is unnecessary. By involving the hardware files, it allows the possibility for conflicts with the FOV files. Possible change suggested below.

File: pydarn/plotting/fan.py [hunk 2]
```diff
@@ -268,6 +270,9 @@ def plot_fov(cls, stid: str, dtime: dt.datetime, ax=None,
...
         beam_corners_aacgm_lats, beam_corners_aacgm_lons = \
             radar_fov(stid, coords='aacgm', date=dtime)
         fan_shape = beam_corners_aacgm_lons.shape
```
Comment (billetd) [inline pydarn/plotting/fan.py:278]:
```suggestion
        fan_shape = beam_corners_aacgm_lons.shape
        
        if ranges == []
            ranges = [0, fan_shape[1]]
```

**Review by billetd — CHANGES_REQUESTED** [review]

File: pydarn/plotting/fan.py (hunk 1 above)
Comment (mts299) [reply pydarn/plotting/fan.py:275]: Well, that is rude? It is when you limit the FOV to being smaller than it really it is.

File: pydarn/plotting/fan.py (hunk 2 above)
Comment (mts299) [reply pydarn/plotting/fan.py:278]: I will use this bandage to replace the actual necessary solution where the radar FOV files need to be consistent with the hardware files.

**Comment by mts299** [comment]
So looks like this issue might be with the hardware files having 225 for our few radars soooo we will wait on discussing this with Kevin.

**Comment by mts299** [comment]
@billetd switched to your solution as mine was unnecessary and just a ban aid. Sorry. 

You can take the lead from here and other problems and defaults in the future.

File: pydarn/plotting/fan.py [hunk 3]
```diff
@@ -27,7 +27,7 @@
 # Third party libraries
 import aacgmv2
 
-from pydarn import PyDARNColormaps, build_scan, radar_fov
+from pydarn import PyDARNColormaps, build_scan, radar_fov, SuperDARNRadars
```
Comment (carleyjmartin) [inline pydarn/plotting/fan.py:30]:
```suggestion
from pydarn import PyDARNColormaps, build_scan, radar_fov
```
Imported but not used

**Review by carleyjmartin — APPROVED** [review]
Tested with:
```python
import matplotlib.pyplot as plt 
import datetime as dt
import pydarn

#grid_file = "/Users/carley/Documents/data/grids/20011220.8000.grid"
grid_file = "/Users/carley/Documents/data/grids/20011220.0800.00.han.grid"
grid_data = pydarn.SuperDARNRead(grid_file).read_grid()

fanplot = pydarn.Grid.plot_grid(grid_data, parameter = 'pwr')

plt.show()
```
for grid plot examples and tested on some old fan plotting code just to check and no issues now Dan fixed the typo in fan_shape. Just one suggestion on an import not used.

**Comment by mts299** [comment]
@billetd is this good to merge?

**Review by billetd — APPROVED** [review]

**Pull request merged**